src/main/ts/components/Chart.tsx

```diff
@@ -49,25 +53,53 @@ export default function Chart() {
     caycStartingDate,
     chartStartingDate,
     chartEndDate,
+    setSelectedProjects,
   } = useData();
 
-  if (isLoading) {
-    return <Spinner />;
-  }
+  const handleLoadProjects = async (q: string, resolve: (options: ProjectOption[]) => void) => {
+    const components = await getProjects({ q });
+    if (!components || components.length === 0) {
+      resolve([]);
+      return [];
+    }
+    const options = [
+      defaultProject,
+      ...components.map(({ key, name }) => {
+        return {
+          value: key,
+          label: name,
+        } as ProjectOption;
+      }),
+    ];
+    resolve(options);
+    return options;
+  };
+
+  const debouncedLoadProjects = React.useRef(
+    debounce((q: string, resolve: (options: ProjectOption[]) => void) => {
+      handleLoadProjects(q, resolve);
+    }, DEBOUNCE_DELAY),
+  );
+
+  const hasData =
+    cumulativeData?.length > 0 &&
+    caycProjectionData?.length > 0 &&
+    chartStartingDate &&
+    caycStartingDate &&
+    !isLoading;
 
   if (hasRequestFailed) {
     return <p>{translate('cayc.request_failed')}</p>;
   }
 
-  if (cumulativeData?.length === 0 || caycProjectionData?.length === 0) {
-    return <p>{translate('cayc.no_data')}</p>;
-  }
-
-  const formatYScale = yScale.tickFormat(undefined, '~s');
-  const issuesDelta = formatYScale(
-    cumulativeData[cumulativeData.length - 1].y -
-      caycProjectionData[caycProjectionData.length - 1].y
-  );
+  const issuesDelta = () =>
+    yScale.tickFormat(
+      undefined,
+      '~s',
+    )(
+      cumulativeData[cumulativeData.length - 1].y -
+        caycProjectionData[caycProjectionData.length - 1].y,
+    );
```

Comment:
Can't you compile this value right here? As it was before?
```suggestion
  const issuesDelta = yScale.tickFormat(
    undefined,
    '~s',
  )(
    cumulativeData[cumulativeData.length - 1].y -
      caycProjectionData[caycProjectionData.length - 1].y,
  );
```

Reply:
I can't, because I don't check for the absence of a selected project before showing the "no data" message. And this `yScale` is undefined where there is no data. That's why I had to transform the function into a thunk and pass the yScale.tickFormat return function directly.